### PR TITLE
Rework Azure Key Vault

### PR DESCRIPTION
**Pull request by rain-on**
The Azure Key Vault would not be loaded if cmdline parameters were
specified via environment variables.

### CODE REVIEW

File: commandline/src/main/java/tech/pegasys/web3signer/commandline/PicoCliAzureKeyVaultParameters.java [hunk 1]
```diff
@@ -18,35 +18,43 @@
 
 public class PicoCliAzureKeyVaultParameters implements AzureKeyVaultParameters {
 
+  @Option(
+      names = {"--azure-vault-enabled"},
+      description =
+          "Set true if Web3signer should try and load all keys in the specified vault "
+              + "(Default: ${DEFAULT-VALUE})",
+      paramLabel = "<ENABLED>")
```
Comment (jframe) [inline commandline/src/main/java/tech/pegasys/web3signer/commandline/PicoCliAzureKeyVaultParameters.java:26]: thinking the paramLabel should be BOOL as that's what have done elsewhere

File: core/src/main/java/tech/pegasys/web3signer/core/Eth2Runner.java [hunk 2]
```diff
@@ -167,8 +168,16 @@ private ArtifactSignerProvider loadSigners(
...
 
-    if (config.getAzureKeyVaultParameters() != null) {
+    if (config.getAzureKeyVaultParameters().isAzureKeyVaultEnabled()) {
       final AzureKeyVaultParameters params = config.getAzureKeyVaultParameters();
+      if (!ObjectUtils.allNotNull(
+          params.getClientlId(),
+          params.getClientSecret(),
+          params.getTenantId(),
+          params.getKeyVaultName())) {
+        throw new InitializationException(
```
Comment (jframe) [inline core/src/main/java/tech/pegasys/web3signer/core/Eth2Runner.java:178]: Will this show up as a nice error message on the cli? Not just a java exception with stacktrace?

File: core/src/main/java/tech/pegasys/web3signer/core/Eth2Runner.java [hunk 3]
```diff
@@ -167,8 +168,16 @@ private ArtifactSignerProvider loadSigners(
             "yaml",
             new YamlSignerParser(List.of(artifactSignerFactory))));
 
-    if (config.getAzureKeyVaultParameters() != null) {
+    if (config.getAzureKeyVaultParameters().isAzureKeyVaultEnabled()) {
       final AzureKeyVaultParameters params = config.getAzureKeyVaultParameters();
+      if (!ObjectUtils.allNotNull(
```
Comment (jframe) [inline core/src/main/java/tech/pegasys/web3signer/core/Eth2Runner.java:173]: Can you add a unit test for this validation check

File: gradle/versions.gradle [hunk 4]
```diff
@@ -97,6 +97,9 @@ dependencyManagement {
 
     dependency 'org.miracl.milagro.amcl:milagro-crypto-java:0.4.0'
 
+    dependency 'com.azure:azure-security-keyvault-secrets:4.2.1'
```
Comment (jframe) [inline gradle/versions.gradle:100]: Are these required?

File: commandline/src/main/java/tech/pegasys/web3signer/commandline/PicoCliAzureKeyVaultParameters.java (hunk 1 above)
Comment (rain-on) [reply commandline/src/main/java/tech/pegasys/web3signer/commandline/PicoCliAzureKeyVaultParameters.java:26]: done

File: core/src/main/java/tech/pegasys/web3signer/core/Eth2Runner.java (hunk 2 above)
Comment (rain-on) [reply core/src/main/java/tech/pegasys/web3signer/core/Eth2Runner.java:178]: yeah, had to do a little refactor to make this come out nicely.

File: gradle/versions.gradle (hunk 4 above)
Comment (rain-on) [reply gradle/versions.gradle:100]: removed.

File: core/src/main/java/tech/pegasys/web3signer/core/Eth2Runner.java (hunk 3 above)
Comment (rain-on) [reply core/src/main/java/tech/pegasys/web3signer/core/Eth2Runner.java:173]: :( At the moment, there are NO unit tests around the Runners - so, can, but it'll be hard...

**Review by jframe — APPROVED** [review]

**Pull request merged**